### ISSUE TITLE
How to track down a MissingBackpressureException?

### ISSUE THREAD

**Issue by Dirk-c-Walter** [ReactiveX/RxJava#3746]
So I'm getting MissingBackpressureExceptions, and I have been trying to litter my code with onBackPressureBuffer calls, which cuts down on their numbers but that doesn't seem like an efficient or streamlined solution. On the other hand a stacktrace like the following tells me nothing about where the problem is.

```
Caused by: rx.exceptions.MissingBackpressureException: null
at rx.internal.operators.OperatorMapNotification$SingleEmitter.offer(OperatorMapNotification.java:165)
at rx.internal.operators.OperatorMapNotification$MapNotificationSubscriber.onNext(OperatorMapNotification.java:98)
at rx.internal.operators.OnSubscribeFromIterable$IterableProducer.slowpath(OnSubscribeFromIterable.java:97)
at rx.internal.operators.OnSubscribeFromIterable$IterableProducer.request(OnSubscribeFromIterable.java:73)
at rx.internal.producers.ProducerArbiter.setProducer(ProducerArbiter.java:126)
at rx.internal.operators.OperatorMapNotification$MapNotificationSubscriber.setProducer(OperatorMapNotification.java:74)
at rx.internal.operators.OnSubscribeFromIterable.call(OnSubscribeFromIterable.java:49)
at rx.internal.operators.OnSubscribeFromIterable.call(OnSubscribeFromIterable.java:32)
at rx.Observable$2.call(Observable.java:162)
at rx.Observable$2.call(Observable.java:154)
at rx.Observable$2.call(Observable.java:162)
at rx.Observable$2.call(Observable.java:154)
at rx.Observable$2.call(Observable.java:162)
at rx.Observable$2.call(Observable.java:154)
at rx.Observable$2.call(Observable.java:162)
at rx.Observable$2.call(Observable.java:154)
at rx.Observable$2.call(Observable.java:162)
at rx.Observable$2.call(Observable.java:154)
at rx.Observable$2.call(Observable.java:162)
at rx.Observable$2.call(Observable.java:154)
at rx.Observable.subscribe(Observable.java:8407) 
at rx.Observable.subscribe(Observable.java:8374)
at rx.observables.BlockingObservable.blockForSingle(BlockingObservable.java:433)
... 8 common frames omitted
```

If there was some indication of exactly where in the chain the problem was that would be helpfull. Instead I just know there is an issue somewhere in my stream, and I have not been able to narrow it down. Are there any techniques or teicks that help with that?

Just in case it helps here is the code generating that exception, though I am more interested in being able to solve the general case.

```
`//'final Collection<String> ids' contains ~500 string ids

rx.Observable
	.from(ids)
	.flatMap(
		(Fnuc1<String, Observable<LoaderValue<V>>>) id -> fetchFromCouchbase(id),
		(Func1<Throwable, Observable<LoaderValue<V>>>) err -> (Observable<LoaderValue<V>>) handleError(err),
		() -> (Observable<LoaderValue<V>>) handleCompletion(),
		100
	)
	.timeout(10000, MILLISECONDS) 
	.onBackpressureBuffer(1000, () -> logOverflow())
	.toList()
	.toBlocking().single();


  private Observable<LoaderValue<V>> fetchFromCouchbase(String id) {
    if(id == null) {
	return Observable.<LoaderValue<V>>empty().onBackpressureDrop();
    }
    return couchbaseBucket.async().get(id, conf.getCouchbaseDocumentImplClass())
	    .cacheWithInitialCapacity(1)
	    .singleOrDefault(null)
	    .map((Func1<Document, LoaderValue<V>>) val -> transform(id, val));
  }
```

handleCompletion and logOverflow just log that the event happened, handleCompletion returns an empty Observable.
handleError logs the error and returns a sentinel value indicating an error occurred.`

**Comment by akarnokd** [comment]
There is a bug in the `mapNotification` operator used by that particular `flatMap` overload that's causing the `MissingBackpressureException`. The fix will be part of 1.1.2, no relase date yet.

**Comment by Dirk-c-Walter** [comment]
I see, thanks for letting me know.
